### PR TITLE
PSTPAY-3927: Email notification templates added for Positive Pay ACH

### PR DESCRIPTION
**Pull request by wromanek**

### CODE REVIEW

File: collections/business/general-notifications/event-general-notifications/pp-ach-upcoming-cut-off-time/pp-ach-upcoming-cut-off-time/email/en_US/engagement-template.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "subject": "Review ACH exceptions by {{{event.cutOffTime}}} today",
+    "body": "Dear {{{user.preferredName}}} {{{user.familyName}}},\n\nYou have exceptions that need to be reviewed by {{{event.cutOffTime}}} today.\n\nIf you do not review them before this cut-off time, they will be {{#eq event.defaultDecision 'PAY' }}paid{{else}}rejected{{/eq}}.\n\nTo review your exceptions:\n\n\t* Log in to Backbase\n\t* Go to ACH Positive Pay\n\t* Click Exceptions-Active\n\nIf you have any questions about this email, please contact [contact details].\n\nRegards,\n\n[Responsible Dept name]"
+}
```
Comment (amorimluzia) [inline collections/business/general-notifications/event-general-notifications/pp-ach-upcoming-cut-off-time/pp-ach-upcoming-cut-off-time/email/en_US/engagement-template.json:4]: where is the email_channel-settings.json file for this another GN?

**Review by amorimluzia — CHANGES_REQUESTED** [review]

**Comment by amorimluzia** [comment]
@wromanek I noticed those GNs are also on Modelbank repo. e.g. https://github.com/baas-devops-reference/engagements-data/blob/master/collections/business/general-notifications/event-general-notifications/pp-ach-exceptions-created/event-general-notifications.json

do you intend to add email channel there as well?

**Comment by wromanek** [comment]
> @wromanek I noticed those GNs are also on Modelbank repo. e.g. https://github.com/baas-devops-reference/engagements-data/blob/master/collections/business/general-notifications/event-general-notifications/pp-ach-exceptions-created/event-general-notifications.json
> 
> do you intend to add email channel there as well? cc: @afsinka-backbase

Yes. Model Bank will get exactly same PR.

File: collections/business/general-notifications/event-general-notifications/pp-ach-exceptions-created/pp-ach-exceptions-created/email/email_channel-settings.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+    "level": "INFO",
+    "targetGroup": "USER",
+    "origin": "PositivePayACH"
+}
```
[the remark this answers is not in the document]
Comment (wromanek) [reply collections/business/general-notifications/event-general-notifications/pp-ach-exceptions-created/pp-ach-exceptions-created/email/email_channel-settings.json:3]: We have tested it only checking the logs of engagement orchestrator. However I have changed it as suggested.

File: collections/business/general-notifications/event-general-notifications/pp-ach-exceptions-created/event-general-notifications.json [hunk 3]
```diff
@@ -27,6 +27,22 @@
                             "locale": "en_US"
                         }
                     ]
+                },
+                {
+                    "channel": "email",
+                    "channelSettings": "pp-ach-exceptions-created/email/email_channel-settings.json",
+                    "isEnabledByDefault": false,
+                    "isManageableByUser": true,
```
[the remark this answers is not in the document]
Comment (wromanek) [reply collections/business/general-notifications/event-general-notifications/pp-ach-exceptions-created/event-general-notifications.json]: Deleted.

File: collections/business/general-notifications/event-general-notifications/pp-ach-upcoming-cut-off-time/event-general-notifications.json [hunk 4]
```diff
@@ -16,7 +16,7 @@
           "channel": "in-app-notification",
           "channelSettings": "pp-ach-upcoming-cut-off-time/in-app-notification/in-app_channel-settings.json",
           "isEnabledByDefault": true,
-          "isManageableByUser": false,
+          "isManageableByUser": true,
```
[the remark this answers is not in the document]
Comment (wromanek) [reply collections/business/general-notifications/event-general-notifications/pp-ach-upcoming-cut-off-time/event-general-notifications.json]: Deleted.

File: collections/business/general-notifications/event-general-notifications/pp-ach-upcoming-cut-off-time/event-general-notifications.json [hunk 5]
```diff
@@ -27,6 +27,21 @@
...
           ]
+        },
+        {
+          "channel": "email",
+          "isEnabledByDefault": false,
+          "isManageableByUser": true,
+          "routeInfo": {
+            "whereTo": "positive-pay-ach-active-exceptions-view",
+            "data": ""
+          },
```
[the remark this answers is not in the document]
Comment (wromanek) [reply collections/business/general-notifications/event-general-notifications/pp-ach-upcoming-cut-off-time/event-general-notifications.json]: Deleted.

File: collections/business/general-notifications/event-general-notifications/pp-ach-upcoming-cut-off-time/event-general-notifications.json [hunk 6]
```diff
@@ -27,6 +27,21 @@
               "locale": "en_US"
             }
           ]
+        },
+        {
+          "channel": "email",
+          "isEnabledByDefault": false,
+          "isManageableByUser": true,
```
[the remark this answers is not in the document]
Comment (wromanek) [reply collections/business/general-notifications/event-general-notifications/pp-ach-upcoming-cut-off-time/event-general-notifications.json]: Deleted.

File: collections/business/general-notifications/event-general-notifications/pp-ach-exceptions-created/event-general-notifications.json [hunk 7]
```diff
@@ -27,6 +27,22 @@
...
+                    "routeInfo": {
+                        "whereTo": "positive-pay-ach-active-exceptions-view",
+                        "data": ""
+                    },
```
[the remark this answers is not in the document]
Comment (wromanek) [reply collections/business/general-notifications/event-general-notifications/pp-ach-exceptions-created/event-general-notifications.json]: Deleted.

File: collections/business/general-notifications/event-general-notifications/pp-ach-upcoming-cut-off-time/pp-ach-upcoming-cut-off-time/email/en_US/engagement-template.json (hunk 1 above)
Comment (wromanek) [reply collections/business/general-notifications/event-general-notifications/pp-ach-upcoming-cut-off-time/pp-ach-upcoming-cut-off-time/email/en_US/engagement-template.json:4]: Added.Added.

**Comment by wromanek** [comment]
@amorimluzia can I merge this PR or should I wait for some approvals?

**Comment by amorimluzia** [comment]
Hey @wromanek please note this email is just an internal example. So you need to configure the appropriate email in order to execute the e2e validation in your application. 

{
    "from": "engage-qa@backbase.com"
}

cc: @afsinka-backbase

**Comment by wromanek** [comment]
> Hey @wromanek please note this email is just an internal example. So you need to configure the appropriate email in order to execute the e2e validation in your application.
> 
> { "from": "[engage-qa@backbase.com](mailto:engage-qa@backbase.com)" }
> 
> cc: @afsinka-backbase

Can't we use the same email? Is it documentd anywhere how can we configure email in order to execute e2e validation?

**Comment by amorimluzia** [comment]
> > Hey @wromanek please note this email is just an internal example. So you need to configure the appropriate email in order to execute the e2e validation in your application.
> > { "from": "[engage-qa@backbase.com](mailto:engage-qa@backbase.com)" }
> > cc: @afsinka-backbase
> 
> Can't we use the same email? Is it documentd anywhere how can we configure email in order to execute e2e validation?

@wromanek  We created this email to Modelbank environments, so on Modelbank it will work and it is already configured in the whitelist. Regarding your application, really depends on the purpose of it. However, if it is a client application I would recommend you to configure a professional email related to the client once you fork the repo or configure it inside the client's application. On public repo you can add a fake email.
cc: @afsinka-backbase @eacarvalho

**Review by amorimluzia — APPROVED** [review]

**Comment by amorimluzia** [comment]
> @amorimluzia can I merge this PR or should I wait for some approvals?

@wromanek I approved your PR, at least one approval is needed. Please just take a look at my another comment related to the email itself. Regarding the other points, I'm fine.

**Comment by wromanek** [comment]
> > > Hey @wromanek please note this email is just an internal example. So you need to configure the appropriate email in order to execute the e2e validation in your application.
> > > { "from": "[engage-qa@backbase.com](mailto:engage-qa@backbase.com)" }
> > > cc: @afsinka-backbase
> > 
> > 
> > Can't we use the same email? Is it documentd anywhere how can we configure email in order to execute e2e validation?
> 
> @wromanek We created this email for Modelbank environments, so on Modelbank it will work and it is already configured in the whitelist. Regarding your application, really depends on the purpose of it. However, if it is a client application I would recommend you to configure a professional email related to the client once you fork the repo or configure it inside the client's application. On public repo you can add a fake email. cc: @afsinka-backbase @eacarvalho

@amorimluzia what I am trying to merge so far is a sample template with sample configuration. It is not customer dedicated implementation. I would leave it as it is so far and later on when anyone will do the customer dedicated implementation will need to change it. But I believe this is obvious since e-mail templates will also need customization.

**Pull request merged**